### PR TITLE
Speedy ANF : bend scalafmt to my will.

### PR DESCRIPTION
**Pull request by nickchapman-da**
Improve the readability of the speedy ANF code:

- Splitting a final continuation parameter into separate parameter list makes output from `scalafmt` much more pleasing!

- Reorder parameters to various defs so that `transform` comes after `k` because `transform` is the continuation which changes most often.

- Inline local defs of `tx1`, `tx2` etc, which were introduced to help `scalac` type check correctly, but it seems are not needed anymore.

This change is a pure refactoring. There is no semantic change.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Anf.scala [hunk 1]
```diff
@@ -271,12 +262,8 @@ private[lf] object Anf {
     *  Note: this wrapping is the reason why we need a "second" CPS transform to
     *  achieve constant stack through trampoline.
     */
-  private[this] def transformExp[A](
-      depth: DepthA,
-      env: Env,
-      exp: SExpr,
-      transform: Tx[SExpr, A],
-      k: K[AExpr, A]): Trampoline[A] =
+  private[this] def transformExp[A](depth: DepthA, env: Env, exp: SExpr, k: K[AExpr, A])(
```
Comment (SamirTalwar-DA) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Anf.scala:265]: Might wanna split `k` out too.

```suggestion
  private[this] def transformExp[A](depth: DepthA, env: Env, exp: SExpr)(k: K[AExpr, A])(
```

**Comment by cocreature** [comment]
Blocked by #6645 which reverts the changes that are being formatted here.

**Comment by cocreature** [comment]
@nickchapman-da I believe this can be closed after the second iteration of ANF has landed.

**Comment by hurryabit** [comment]
@nickchapman-da If there's anything to be resurrected from this PR, can you please take care of it?

**Comment by nickchapman-da** [comment]
> @nickchapman-da If there's anything to be resurrected from this PR, can you please take care of it?

It already got salvaged into the partial ANF which landed

**Pull request closed without merging**